ci/pkl-pipelines/pay-deploy/deploy-to-production.pkl

```diff
@@ -0,0 +1,834 @@
+amends "package://pkg.pkl-lang.org/github.com/alphagov/pkl-concourse-pipeline/pkl-concourse-pipeline@0.0.3#/Pipeline.pkl"
+
+import "package://pkg.pkl-lang.org/github.com/alphagov/pkl-concourse-pipeline/pkl-concourse-pipeline@0.0.3#/TaskConfig.pkl"
+
+import "../common/pipeline_self_update.pkl"
+import "../common/shared_resources.pkl"
+import "../common/shared_resources_for_metrics.pkl"
+import "../common/shared_resources_for_slack_notifications.pkl"
+
+local class PayApp {
```

Comment:
After this PR it would be good to refactor this and deploy-to-perf and extract some of this stuff which feels shared out into something shared, but I think for this PR it's ok here